javalib/src/main/scala/java/util/ArrayDeque.scala

```diff
@@ -0,0 +1,231 @@
+// Ported from Scala.js.
+// Also contains original work for Scala Native.
+
+package java.util
+
+/// ScalaNative Porting Note:
+///
+///     * Ported, with thanks & gratitude, from Scala.js ArrayDeque.scala
+///       commit 9DC4D5b, dated 2018-10-12.
+///       Also contains original work for Scala Native.
+///
+///     * Changes in Scala.js original commit E07F99D, dated 2019-07-30
+///       were considered on 2020-05-19. The Scala.js change to
+///       ArrayDeque.scala were to Objects.equals() in a 3 places,
+///       contains(), removeFirstOccurrence(), removeLastOccurrence().
+///       Existing unit-tests for those three methods show that no
+///       corresponding change is needed here.  indexOf() and contains()
+///       are documented as defined in terms of "==". That operator should
+///       do and is doing a comparison equivalent to Object.equals().
```

Comment:
No, `==` is not equivalent to `Object.equals()`. `==` uses `__scala_==` whereas `Object.equals()` uses `equals`. I'm sure by now you know the difference between those, since you've recently looked at `__scala_==`.

However, here you don't need `Object.equals()` because the above methods are defined in terms of `inner.{contains,indexOf,lastIndexOf}`. And since `inner` is a `java.util.ArrayList`, its methods use the semantics of `Object.equals()`. This is not the case in the Scala.js variant, because there `inner` is a `js.Array`, whose methods are Scala collection methods, with the semantics of `==`.

Reply:
As always, thank you for the correction.   I hope my incorporation and slight paraphrase
of you explanation is acceptable use. I think it provides at least a clue for the 
question in 6 months: OK, why is no change needed?

Alternately, I can just drop everything after "No corresponding change  is needed here".